src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -122,27 +124,23 @@ public class UTIL_ACCT_Naming {
         String sortByFirstName = 'FirstName';
 
         //Build dynamic query string
-        String dynamicSoql = 'SELECT id, RecordTypeId, ';
+        String dynamicSoql = 'SELECT Id, RecordTypeId, ';
         dynamicSoql += '(SELECT Id, AccountId,' +
             'Account.RecordTypeID, ' +
             'Account.Primary_Contact__c, Account.Name,' +
             'firstname, lastname, OwnerId, Salutation,' +
             'MailingStreet, MailingCity, MailingState, ' +
             'MailingPostalCode, MailingCountry, MailingLatitude, ' +
             'MailingLongitude, OtherStreet, OtherCity, OtherState, ' +
-            'OtherPostalCode, OtherCountry, OtherLatitude, OtherLongitude, ';
+            'OtherPostalCode, OtherCountry, OtherLatitude, OtherLongitude, ' +
+            'Phone, Fax';
 
         if (ADDR_Addresses_UTIL.isStateCountryPicklistsEnabled) {
-            dynamicSoql += 'MailingCountryCode, MailingStateCode, OtherCountryCode, OtherStateCode, ';
-        }
-
-        if (defaultRecTypeId == hhAccountRecordTypeId) {
-            dynamicSoql += 'Phone, Fax FROM Contacts WHERE Exclude_from_Household_Name__c != true)';
-        } else {
-            dynamicSoql += 'FROM Contacts)';
+            dynamicSoql += ', MailingCountryCode, MailingStateCode, OtherCountryCode, OtherStateCode';
         }
 
-        dynamicSoql += 'FROM Account WHERE Id IN :accIdsToRename';
+        dynamicSoql += ' FROM Contacts WHERE Exclude_from_Household_Name__c != true AND Deceased__c != true)';
```

Comment:
> This is not inside the if statement that checked for Household record type before. But we are still checking the `Exclude_from_Household_Name__c` field. What happens if that field doesn't have a value of if it's another type of account? Are we planning on using this field for other type of accounts?

@ceiroa -- We only want contacts included when they need to be considered for naming the account in question. I'm relying here on the hopes that if we are autonaming an admin account that it will only have one contact and in fact the exclude from naming better not be true. I feel like there might be a better way but this is a good stop-gap since we do have a production issue.

Reply:
From discussion - we need to test with state and country picklists enabled and disabled (write unit tests if possible).

Reply:
Tested happy path with admin account and hh account